### ISSUE TITLE
Fortran 95 should allow the optional :: in external statements.

### ISSUE THREAD

**Issue by jme52** [camfort/fortran-src#60]
The Fortran 95 standard (https://wg5-fortran.org/N1151-N1200/N1191.pdf) allows for an optional double colon `::` as part of the EXTERNAL statement (see page 217 of the pdf):

> R1208 _external-stmt_ **is** EXTERNAL [ :: ] _external-name-list_

As far as I can see, this is not allowed yet by fortran-src 0.2.1.1.

Example:
```
program ex
  implicit none

  integer :: myfunc
  external :: myfunc

  write(*,*) myfunc(3)

end program
```

Current camfort 0.905 output for the previous example:
```
$ stack exec camfort count ex.f90 
Counting variable declarations in 'ex.f90'
ex.f90, 5:14: ex.f90: parsing failed. 
Last parsed token: TDoubleColon (5:12)-(5:13).
```

**Comment by mrd** [comment]
Fixed in 921da4ee3e825740b37162cc4ec5db4baad4477a assuming you specify -F95 or use file extension .f95.

**Comment by jme52** [comment]
Thank you for the fix!

Concerning the `.f95` file extension, same comment as in https://github.com/camfort/fortran-src/issues/62#issuecomment-459039921 applies.
